### PR TITLE
Conditionally use cacheRefresh for memberships

### PR DESCRIPTION
**Pull request by rbennettcw**
## Link to Issue
Closes: #5804

## Description of Changes
- Uses cached balances for refresh triggered by group requirements update + gated actions
- Uses fresh balances for single-user refresh (on page refresh)

## Test Plan
CA testing this on demo is not feasible, but locally is doable.

The goal of the test is to confirm that gated actions use cached data while page refreshes use fresh data:
- Set env `MEMBERSHIP_REFRESH_TTL_SECONDS` = 15
- Set env `TBC_BALANCE_TTL_SECONDS` = 1
- As a non-admin, go to a gated thread in which you are not the author
- Truncate the Memberships table (`DELETE FROM “Memberships”`)
- Perform these steps quickly in less than 10 seconds
    - Upvote the thread to create a reaction
    - Query all memberships (`SELECT * FROM “Memberships”`)– confirm that 1 exists, take note of the `last_checked` time
    - Click the upvote button multiple times to create the reaction multiple times
    - Query all memberships (`SELECT * FROM “Memberships”`)– confirm that 1 exists, and the `last_checked` column is the SAME as before
- Perform these steps quickly
    - Refresh the page
    - Query all memberships (`SELECT * FROM “Memberships”`)– confirm that at least 1 exists, take note of the `last_checked` time
    - Confirm that every time the page refreshes, the `last_checked` time is DIFFERENT from before

## Deployment Plan
N/A

## Other Considerations
N/A

### CODE REVIEW

File: packages/commonwealth/server/util/requirementsModule/validateTopicGroupsMembership.ts
Comment (rbennettcw) [inline packages/commonwealth/server/util/requirementsModule/validateTopicGroupsMembership.ts:1]: This file handles the requirements check for gated actions.

The old code did not check DB memberships– it used the TBC directly to perform the requirements check.

Now, it uses the shared `refreshMembershipsForAddress` function which checks the membership record before hitting the TBC.

**Comment by jnaviask** [comment]
@rbennettcw to fix merge conflicts, and also can we use #5985 to create an actual test plan here, so we can QA?

**Review by jnaviask — APPROVED** [review]
Seems OK to me.

**Review by timolegros — APPROVED** [review]
Untested but looks fine.

**Comment by rbennettcw** [comment]
Added test plan.

It's only feasible to test locally IMO, since the end-user won't see any perceptible difference and gated actions don't expose membership data via the API payload.

**Pull request merged**